ikarus/finiteelements/mechanics/nonlinearelastic.hh

```diff
@@ -41,27 +54,22 @@ namespace Ikarus {
  * \tparam useEigenRef A boolean flag indicating whether to use Eigen references.
  * \tparam useFlat A boolean indicating if the type of the underlying basis is of the flat or the untouched version.
  */
-template <typename BH, typename MAT, typename FER = FERequirements<>, bool useEigenRef = false, bool useFlat = true>
-class NonLinearElastic : public FEBase<BH, useFlat, FER, useEigenRef>,
-                         public Volume<NonLinearElastic<BH, MAT, FER, useEigenRef, useFlat>,
-                                       typename FEBase<BH, useFlat, FER, useEigenRef>::Traits>,
-                         public Traction<NonLinearElastic<BH, MAT, FER, useEigenRef, useFlat>,
-                                         typename FEBase<BH, useFlat, FER, useEigenRef>::Traits>
+template <typename PreFE, typename FE, typename PRE>
```

Comment:
Why this has an additional template argument `PRE` compared to other finite elements? It can be removed?

Reply:
No because PREFE does not contain the material information and FE is an incomplete type we need an accesable time to deduce the material type.

Reply:
Ah okay, may be then the other elements can also have `PRE` passed, but this can also be done in a separate house keeping PR